modules/fundamental/src/vulnerability/endpoints/mod.rs

```diff
@@ -93,44 +92,22 @@ pub async fn get(
         ("id", Path, description = "ID of the vulnerability")
     ),
     responses(
-        (status = 200, description = "Specified vulnerability", body = VulnerabilityDetails),
-        (status = 404, description = "Specified vulnerability not found"),
+        (status = 501, description = "Method not yet implemented"),
     ),
 )]
 #[delete("/v2/vulnerability/{id}")]
+#[deprecated(
+    since = "0.2.19",
+    note = "This endpoint is deprecated and will be removed in a future version."
+)]
 /// Delete vulnerability
 pub async fn delete(
-    state: web::Data<VulnerabilityService>,
-    db: web::Data<Database>,
-    id: web::Path<String>,
+    _state: web::Data<VulnerabilityService>,
+    _db: web::Data<Database>,
+    _id: web::Path<String>,
     _: Require<DeleteVulnerability>,
 ) -> Result<impl Responder, Error> {
-    let tx = db.begin().await?;
-
-    let vuln = state
-        // we ignore deprecated advisories, as we delete the vulnerability anyway.
-        .fetch_vulnerability(
-            &id,
-            trustify_module_ingestor::common::Deprecation::Ignore,
-            &tx,
-        )
-        .await?;
-
-    if let Some(vuln) = vuln {
-        let rows_affected = state
-            .delete_vulnerability(&vuln.head.identifier, &tx)
-            .await?;
-        match rows_affected {
-            0 => Ok(HttpResponse::NotFound().finish()),
-            1 => {
-                tx.commit().await?;
-                Ok(HttpResponse::Ok().json(vuln))
-            }
-            _ => Err(Internal("Unexpected number of rows affected".into())),
-        }
-    } else {
-        Ok(HttpResponse::NotFound().finish())
-    }
+    Ok(HttpResponse::NotImplemented().finish())
```

Comment:
I'm not against this. But "deprecation" would mean to me, that it is still there, but will be removed in a future version.

This effectively removes it (the functionality) today.

Maybe we can be bit more clear in the description of this: it never really worked, we disabled it for now?